### PR TITLE
Adding SeqLib as an alternative library for handling Bam files in filterBam.

### PR DESCRIPTION
**Pull request by hmehlan**
- With the CPPFLAG "USE_SEQLIB=true" you can switch from the Bamtools library to the SeqLib library in  [auxprogs/filterBam/src/Makefile](https://github.com/Gaius-Augustus/Augustus/blob/a0149e0b1ec98304839fdecc097b0a1973a6d54b/auxprogs/filterBam/src/Makefile#L6).

- Nathan Weeks SeqLib implementation needed only 37% of the time for a 305Mb Bam file compared to BamTools (or 26 % using 3 threads).
- By setting the "-O3" compiler option the runtime dropped to 80% for both libraries.
- By performance refactoring and "-O3" the runtime using the Bamtools library dropped to 45% of the initial value.
- The runtime with the SeqLib library did not sink so strongly, since in Nathans code some simplifications led to wrong results and had to be fixed:
-- currently the implementation with the SeqLib library needs about 75% of the time compared to current Bamtools (or 33% to initial Bamtools), 
-- or by using 3 threads 40% compared to Bamtools (or 18% to initial Bamtools).

### CODE REVIEW

File: auxprogs/filterBam/src/filterBam.cc [hunk 1]
```diff
@@ -11,11 +11,6 @@
 	Last modified: 13-April-2012
```
Comment (MarioStanke) [inline auxprogs/filterBam/src/filterBam.cc:11]: update or delete line

File: auxprogs/filterBam/src/headers/bamseqlibaccess.hh [hunk 2]
```diff
@@ -0,0 +1,179 @@
+/**	read, write and access BAM files through bamtools library */
```
Comment (MarioStanke) [inline auxprogs/filterBam/src/headers/bamseqlibaccess.hh:1]: SeqLib instead of bamtools?

**Review by MarioStanke — APPROVED** [review]
looks good as far as I have comprehended it. Good to be merged from my perspective after the minor changes.

File: auxprogs/filterBam/README.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+filterBAM is a C++ routine for filtering BAM alignment files; it is based on [filterPSL.pl](../../scripts/filterPSL.pl), a Perl script written
+for the cleaning of PSL alignment files.
+
+
+filterBam only accepts as input BAM files that have been previously sorted by 'query name'.
+
+# DEPENDENCIES
+
+1. Bamtools' API **OR** SeqLib API  is required for compilation.
```
[the remark this answers is not in the document]
Comment (hmehlan) [reply auxprogs/filterBam/README.md]: done

File: auxprogs/filterBam/src/BamSeqLibAccess.cc [hunk 4]
```diff
@@ -0,0 +1,273 @@
+/**	read, write and access BAM files through bamtools library */
```
[the remark this answers is not in the document]
Comment (hmehlan) [reply auxprogs/filterBam/src/BamSeqLibAccess.cc]: done

File: auxprogs/filterBam/src/filterBam.cc (hunk 1 above)
Comment (hmehlan) [reply auxprogs/filterBam/src/filterBam.cc]: deleted

File: auxprogs/filterBam/src/headers/bamseqlibaccess.hh (hunk 2 above)
Comment (hmehlan) [reply auxprogs/filterBam/src/headers/bamseqlibaccess.hh]: done

File: auxprogs/filterBam/src/filterBam.cc [hunk 5]
```diff
@@ -344,31 +288,38 @@ int main(int argc, char *argv[])
...
+		    useEqualSigns = true;
+		    checkForEqualSigns = false;
+		}
+		else if (al->getTagData("NM", editDistance)) {
+			// "NM" - Edit distance tag, which records the Levenshtein distance between the read and the reference.
 			percId = (float)100*(qLength-editDistance)/qLength;  
+		  } else {
+			percId = 100.0f;
+			if (!is_NM_WarningPrinted) {
+				std::cout << "WARNING: z alignments (x%) were missing the NM tag, and could therefore "
```
Comment (MarioStanke) [inline auxprogs/filterBam/src/filterBam.cc:319]:
```suggestion
				std::cout << "WARNING: Some alignments were missing the NM tag, and could therefore "
```

**Review by MarioStanke — APPROVED** [review]

**Comment by nathanweeks** [comment]
For the ~831G BAM file mentioned in #297, the line count emitted in the "processed line <integer>" message wrapped & started emitting negative numbers, indicating that an integral type with a larger range than an `int` would be beneficial
https://github.com/Gaius-Augustus/Augustus/blob/b8f0bad532b1606b75221864911ceb6a3c8565e0/auxprogs/filterBam/src/filterBam.cc#L91

**Comment by MarioStanke** [comment]
Thanks, Nathan, I will make this `unsigned long `long` to be sure.

**Pull request merged**